packages/parser/lib/css_lexer.re

```diff
@@ -928,3 +933,51 @@ let from_string = string => {
 
   read([]);
 };
+
+let tokenize = input => {
+  let buffer = Sedlexing.Utf8.from_string(input);
+  let rec from_string = acc => {
+    switch (get_next_tokens_with_location(buffer)) {
+    | (Parser.EOF, _, _) => []
+    | token => [token, ...from_string(acc)]
+    };
+  };
+
+  try(Ok(from_string([]))) {
+  | exn => Error(Printexc.to_string(exn))
+  };
+};
+
+let render_token =
+  fun
+  | Parser.EOF => ""
+  | t => Tokens.token_to_debug(t);
+
+let position_to_string = pos =>
+  Printf.sprintf(
+    "[%d,%d+%d]",
+    pos.Lexing.pos_lnum,
+    pos.Lexing.pos_bol,
+    pos.Lexing.pos_cnum - pos.Lexing.pos_bol,
+  );
+
+let debug_token = ((token, loc_start, loc_end)) => {
+  /* (loc: Ppxlib.Location.t, token: Tokens.token) => { */
+  let pos_start = position_to_string(loc_start);
+  let pos_end = position_to_string(loc_end);
+  Printf.sprintf(
+    "%s %s..%s",
+    Tokens.token_to_debug(token),
+    pos_start,
+    pos_end,
+  );
+};
+
+let to_string = tokens =>
```

Comment:
I think this should be defined in the file it is called in until there is a need to have it here.